### PR TITLE
[BUGFIX] Skip Snowflake FDS tests for User Forks

### PR DESCRIPTION
**Pull request by Shinnnyshinshin**
- [ ] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [ ] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [ ] Code is linted - run `invoke lint` (uses `black` + `ruff`)
- [ ] Appropriate tests and docs have been updated

For more information about contributing, see [Contribute](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: tests/datasource/fluent/integration/test_sql_datasources.py [hunk 1]
```diff
@@ -197,6 +198,17 @@ def databricks_sql_ds(context: EphemeralDataContext) -> DatabricksSQLDatasource:
     return ds
 
 
+def snowflake_creds_populated():
+    if os.getenv("SNOWFLAKE_CI_USER_PASSWORD") in [None, ""] or os.getenv(
+        "SNOWFLAKE_CI_ACCOUNT"
+    ) in [None, ""]:
+        return False
+    return True
```
Comment (Kilo59) [inline tests/datasource/fluent/integration/test_sql_datasources.py]:
```suggestion
    if os.getenv("SNOWFLAKE_CI_USER_PASSWORD") or os.getenv(
        "SNOWFLAKE_CI_ACCOUNT"
    ):
        return True
    return False
```
Python's built-in truthy evaluation will give you want you want here.

Comment (Shinnnyshinshin) [reply tests/datasource/fluent/integration/test_sql_datasources.py]: Thank you @Kilo59

File: tests/datasource/fluent/integration/test_sql_datasources.py [hunk 2]
```diff
@@ -198,7 +199,18 @@ def databricks_sql_ds(context: EphemeralDataContext) -> DatabricksSQLDatasource:
...
-def snowflake_ds(context: EphemeralDataContext) -> SnowflakeDatasource:
+def snowflake_creds_populated() -> bool:
+    if os.getenv("SNOWFLAKE_CI_USER_PASSWORD") or os.getenv("SNOWFLAKE_CI_ACCOUNT"):
+        return True
+    return False
+
+
+@pytest.fixture
+def snowflake_ds(
+    context: EphemeralDataContext, snowflake_creds_populated
```
Comment (Kilo59) [inline tests/datasource/fluent/integration/test_sql_datasources.py]:
```suggestion
    context: EphemeralDataContext, snowflake_creds_populated: bool
```

File: tests/datasource/fluent/integration/test_sql_datasources.py [hunk 3]
```diff
@@ -198,7 +199,18 @@ def databricks_sql_ds(context: EphemeralDataContext) -> DatabricksSQLDatasource:
 
 
 @pytest.fixture
-def snowflake_ds(context: EphemeralDataContext) -> SnowflakeDatasource:
+def snowflake_creds_populated() -> bool:
+    if os.getenv("SNOWFLAKE_CI_USER_PASSWORD") or os.getenv("SNOWFLAKE_CI_ACCOUNT"):
+        return True
+    return False
```
Comment (Kilo59) [inline tests/datasource/fluent/integration/test_sql_datasources.py:205]: `non-blocking`
You can also build the skip directly into the fixture.

**Review by Kilo59 — APPROVED** [review]
I could easily see this logic result in us skipping snowflake tests and not knowing about it.
We should think about moving this and other creds needing test to their own job/gh action.
But that can be a followup.

**Comment by Shinnnyshinshin** [comment]
> I could easily see this logic result in us skipping snowflake tests and not knowing about it. We should think about moving this and other creds needing test to their own job/gh action. But that can be a followup.

good point. Maybe something similar to https://github.com/great-expectations/great_expectations/blob/614939f945fc88dd97181f829ae4203662c524da/.github/workflows/ci.yml#L208

**Pull request merged**